src/workers/personalizedDigestEmail.ts

```diff
@@ -91,45 +96,75 @@ const getPostsTemplateData = ({ posts }: { posts: TemplatePostData[] }) => {
         'email',
         'digest',
       ),
+      post_upvotes: post.upvotes || 0,
+      post_comments: post.comments || 0,
+      post_summary: post.summary,
+      post_read_time: post.readTime,
+      post_views: post.views || 0,
       source_name: post.sourceName,
       source_image: post.sourceImage,
     };
   });
 };
 
-const getEmailVariation = async (
-  variation: number,
-  firstName: string,
-  dayName: string,
-  posts: TemplatePostData[],
-): Promise<Partial<MailDataRequired>> => {
-  const defaultPreview = `Every ${dayName}, we'll send you five posts you haven't read. Each post was carefully picked based on topics you love reading about. Let's get to it!`;
+const getEmailVariation = async ({
+  personalizedDigest,
+  posts,
+  user,
+  feature,
+  currentDate,
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+  posts: TemplatePostData[];
+  user: User;
+  feature: typeof features.personalizedDigest.defaultValue;
+  currentDate: Date;
+}): Promise<Partial<MailDataRequired>> => {
+  const [dayName] = Object.entries(DayOfWeek).find(
+    ([, value]) => value === personalizedDigest.preferredDay,
+  );
+  const userName = user.name?.trim().split(' ')[0] || user.username;
   const data = {
     day_name: dayName,
-    first_name: firstName,
+    first_name: userName,
     posts: getPostsTemplateData({ posts }),
+    date: format(currentDate, 'MMM d, yyyy'),
+    user: {
+      username: user.username,
+      image: user.image,
+      reputation: user.reputation,
+    },
```

Comment:
Using nested object for `user` entity properties. We can probably also apply it to `post` to avoid whole `post_` schema but for backward compatibility let's leave that part as is for now.